File: ffi/ffi_example4/ffi_example4.c
#include <stdio.h>

void function1(void)
{
    puts("Function1!");
}

void function2(int x)
{
    printf("Function2! %d\n", x);
}

void function3(int x, int y)
{
    printf("Function3! %d\n", x+y);
}

void function4(float x)
{
    printf("Function4! %7.5f\n", 1.f/x);
}

void function5(char c)
{
    printf("Function5! %c\n", c);
}

File: ffi/ffi_example4/ffi_example4.lua
-- Ctvrty demonstracni priklad vyuzivajici knihovnu FFI

local ffi = require("ffi")

local lib = ffi.load("ffi_example4")

-- Definice ceckovskych funkci ze standardni knihovny

ffi.cdef[[
void function1(void);
void function2(int);
void function3(int, int);
void function4(float);
void function5(char);
]]

-- Zavolani ceckovskych funkci
lib.function1()
lib.function2(42)
lib.function3(1,2)
lib.function4(3)
lib.function5(97)

